packages/dev/core/src/Gizmos/scaleGizmo.ts

```diff
@@ -176,17 +186,23 @@ export class ScaleGizmo extends Gizmo implements IScaleGizmo {
         this.zGizmo = new AxisScaleGizmo(new Vector3(0, 0, 1), Color3.Blue().scale(0.5), gizmoLayer, this, thickness);
 
         // Relay drag events
-        [this.xGizmo, this.yGizmo, this.zGizmo, this.uniformScaleGizmo].forEach((gizmo) => {
-            gizmo.dragBehavior.onDragStartObservable.add(() => {
-                this.onDragStartObservable.notifyObservers({});
-            });
-            gizmo.dragBehavior.onDragObservable.add(() => {
-                this.onDragObservable.notifyObservers({});
-            });
-            gizmo.dragBehavior.onDragEndObservable.add(() => {
-                this.onDragEndObservable.notifyObservers({});
-            });
-        });
+        const dragStartListener = () => {
+            this.onDragStartObservable.notifyObservers({});
+        };
+        const dragListener = () => {
+            this.onDragObservable.notifyObservers({});
+        };
+        const dragEndListener = () => {
+            this.onDragEndObservable.notifyObservers({});
+        };
+
+        const allGizmos = this.allGizmos;
+        for (let i = 0; i < allGizmos.length; i++) {
+            const gizmo = allGizmos[i];

```

Comment:
From this point on, there are many loops modeled in this way. Consider simply `for (const gizmo of this.allGizmos)`, for the sake of readability.